atc/configwarning.go

```diff
@@ -1,6 +1,46 @@
 package atc
 
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
 type ConfigWarning struct {
 	Type    string `json:"type"`
 	Message string `json:"message"`
 }
+
+type InvalidIdentifierError struct {
+	Identifier string
+	Context    []string
+}
+
+func (e *InvalidIdentifierError) Error() string {
+	return fmt.Sprintf("'%s' is not a valid identifier", e.Identifier)
+}
+
+func (e *InvalidIdentifierError) ConfigWarning() ConfigWarning {
+	warning := ConfigWarning{
+		Type:    "invalid_identifier",
+		Message: e.Error(),
+	}
+
+	if e.Context != nil {
+		warning.Message = fmt.Sprintf("%s: %s", strings.Join(e.Context, ""), e.Error())
+	}
+
+	return warning
+}
+
+var validIdentifiers = regexp.MustCompile(`^\p{L}[\p{L}\d\-.]*$`)
```

Comment:
The [RFC](https://github.com/concourse/rfcs/blob/master/040-valid-identifiers/proposal.md#proposal) mentions that

> All letters will be converted to lowercase. This is to enforce case-insensitivity and to present a consistent UI/UX while still supporting languages that don't have any casing (e.g. Japanese).

We should probably just emit a warning for now. 

@vito in the future, do you see this as the CLI automatically converting the case and logging a warning; or just fail outright?

Reply:
Isn't `\p{L}` short for `[\p{Ll}\p{Lu}\p{Lt}\p{Lm}\p{Lo}]`? If we remove `\p{Lu}` from that we would have valid identifier be lowercase only and support languages without case. Then there's no need for auto-converting case, right?

It's not pretty, but I think `^[\p{Ll}\p{Lt}\p{Lm}\p{Lo}][\p{Ll}\p{Lt}\p{Lm}\p{Lo}\d\-.]*$` would do that...

Reply:
@mouellet Neat! Yeah, that works as expected. I'll amend the RFC with this as well.